packages/itmat-ui-react/src/components/lxd/lxd.instance.text.console.tsx

```diff
@@ -1,234 +1,179 @@
 
-        onConnectionClose();
-        try {
-            // exec
-            console.log('Attempting to fetch the LXD console WebSocket for:', instanceName);
-            console.log('status:', status);
-
-            const response = await fetch(`/lxd/console?c=${instanceName}`, {
-                method: 'POST',
-                headers: { 'Content-Type': 'application/json' },
-                body: JSON.stringify({ 'wait-for-websocket': true, 'type': 'console' })
-            });
-
-            if (!response.ok) throw new Error('Failed to initiate console session');
-
-            const result = await response.json();
-            const baseUrl = new URL(window.location.href);
-            baseUrl.protocol = baseUrl.protocol === 'https:' ? 'wss:' : 'ws:';
-            baseUrl.pathname = '/rtc';
-            const dataUrl = `${baseUrl.href}?t=d&o=${result.operationId}&s=${result.operationSecrets['0']}`;
-
-            const ws = new WebSocket(dataUrl);
-
-            ws.onopen = () => {
-                console.log('Data WebSocket connection established');
-                setLoading(false);
-                dispatch({ type: ActionTypes.OPEN, socket: ws });
-            };
-
-            ws.binaryType = 'arraybuffer';
-
-            ws.onmessage = (message) => {
-                console.log('message from backend:', message.data);
-                const decodedMessage = textDecoder.decode(new Uint8Array(message.data));
-                xtermRef.current?.terminal.write(decodedMessage);
-            };
-
-            ws.onclose = () => {
-                console.log('WebSocket connection closed');
-                setLoading(false);
-                dispatch({ type: ActionTypes.CLOSED });
-                onConnectionClose();
-            };
-
-            ws.onerror = (error) => {
-                console.error('WebSocket error:', error);
-                setLoading(false);
-                dispatch({ type: ActionTypes.CLOSED });
-                onConnectionClose();
-            };
-        } catch (error) {
-            console.error('Connection initiation failed:', error);
+        if (result.error) {
             setLoading(false);
-            dispatch({ type: ActionTypes.CLOSED });
-            onConnectionClose();
-        }
-    }, [instanceName, connectSignal, status, onConnectionClose, fetchConsoleLogBuffer]);
-
-    const handleResize = useCallback(() => {
-        if (xtermRef.current) {
-            const terminalElement = xtermRef.current.terminal.element;
-            updateMaxHeight(terminalElement, 10); // You might need to adjust the offset value
-            fitAddon.fit(); // This will fit the terminal within the new dimensions
-            xtermRef.current?.terminal.focus();
+            throw new Error(`Failed to initiate console session: ${result.error}`);
         }
-    }, [xtermRef, fitAddon]);
 
+        const baseUrl = new URL(window.location.href);  //TODO import  from the common module
```

Comment:
Be careful with using `windows.location.href`, as discussed before this may not work in proxied environments.